### ISSUE TITLE
Allow for identifying 404 vs 405

### ISSUE THREAD

**Issue by absolux** [delvedor/find-my-way#75]
Whenever the router fails to match a request, `.find()` returns `null` for two reasons:

- Method mismatch (405)
- Path mismatch (404)

Is there a way to differentiate between the two in my handler to send the right HTTP error ?

Thanks

**Comment by delvedor** [comment]
No there is not and I'm not going to support it at the moment because it will cost on the overall performances.

We can discuss what does it mean "right HTTP error". I my opinion 404 is correct, because there are not routes that handles the given method.
The 405 error (which by spec is `Method Not Allowed`) should be handled before the routing.
```js
const http = require('http')
const router = require('find-my-way')()
const supportedMethods = ['GET', 'POST']

http.createServer((req, res) => {
  if (supportedMethods.indexOf(req.method) === -1) {
    res.statusCode = 405
    res.end('Method Not Allowed')
  } else {
    router.lookup(req, res)
  }
}).listen(3000)
```

**Comment by delvedor** [comment]
> I agree, but there are cases when a method is allowed for specific routes but not for others

Yes, but this is based on your applications structure and needs, again, for this specific case you should add a check *before* run the router.

This is a router and not an entire framework, this library should call a function based on two strings, the method and the url. Every other logic, such as "auth routes should now allow other methods than..." is out of the scope of the router :)

**Comment by absolux** [comment]
> This is a router and not an entire framework

I'm using a wrapper around `find-my-way` which composes multiple handlers into a single function, and I use `find` instead of `lookup` to find the appropriate route handler, as below:
```js
var { method, url } = ctx.req
var found = this._tree.find(method, _sanitize(url))  // <= I'm using `find` not `lookup`

if (!found) {
  ctx.error = createError(404, `Route not found for ${method} ${url}`)

  return this._throw(ctx)
}

// and here I want to trigger 405 error if the node has the route but not with the requested method

// add url params to the context
ctx.params = found.params || {}

// invoke the route handler
found.handler(ctx)
```
Unfortunately, `find` doesn't return a `Node` instance, but a `handle` object or `null` otherwise.
In my case, I need `findNode(path)` which returns a node instance or null, then I can use `node.getHandler(method`) to get the handle object or undefined.

**Comment by delvedor** [comment]
> I need findNode(path) which returns a node instance or null

Usually a tree structure returns the data stored in a Node and not the Node itself, the node contains way more data which the user should not work with.

> then I can use node.getHandler(method) to get the handle object or undefined.

It will not work in any case, as you can see [here](https://github.com/delvedor/find-my-way/blob/933b63fc23c59c9cc84924308478d28587102108/node.js#L172-L181) the internal handlers object contains all HTTP methods with a default value of `null`.
